compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -224,13 +225,25 @@ impl<'local, 'interner> Interpreter<'local, 'interner> {
     fn get_function_body(&mut self, function: FuncId, location: Location) -> IResult<ExprId> {
         let meta = self.elaborator.interner.function_meta(&function);
         match self.elaborator.interner.function(&function).try_as_expr() {
-            Some(body) => Ok(body),
+            Some(body) => {
+                // Even if the body exists, check if it had errors during elaboration
+                if self.elaborator.functions_with_errors.contains(&function) {
+                    return Err(InterpreterError::SkippedDueToEarlierErrors);
+                }
+                Ok(body)
+            }
             None => {
                 if matches!(&meta.function_body, FunctionBody::Unresolved(..)) {
                     self.elaborate_in_function(None, None, |elaborator| {
                         elaborator.elaborate_function(function);
                     });
 
+                    // After elaboration, check if there were errors before trying to get the body
+                    if self.elaborator.functions_with_errors.contains(&function) {
+                        return Err(InterpreterError::SkippedDueToEarlierErrors);
+                    }
+
+                    // Recursive call - this will now hit the Some(body) branch
```

Comment:
Given that this will hit the `Some` branch, maybe we can avoid checking if the function errored here.

Reply:
Yeah it would reduce code, but I was also thinking it could provide an early return. I don't feel super strongly about keeping this check.